File: maple/maple-compy-web/src/main/webapp/js/service/compyService.js
//服务层
app.service('compyService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../compy/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../compy/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../compy/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../compy/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../compy/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../compy/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../compy/search.do?page='+page+"&rows="+rows, searchEntity);
	}
    //更改状态
    this.updateStatus=function(id,status){
        return $http.get('../compy/updateStatus.do?id='+id+'&status='+status);
    }

    //更改图片
    this.updatePicpath=function(pic){
        return $http.get('../compy/updatePic.do?pic='+pic);
    }
});

File: maple/maple-type-interface/src/main/java/com/maple/service/CompyService.java
package com.maple.service;
import java.util.List;
import com.maple.pojo.Compy;

import entity.PageResult;
/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface CompyService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Compy> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Compy compy);
	
	
	/**
	 * 修改
	 */
	public void update(Compy compy);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Compy findOne(int id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(int[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Compy compy, int pageNum, int pageSize);

	/**
	 * 更新审核状态
	 * @param id
	 * @param status
	 */
	public void updateStatus(int id,int status) ;


	public Compy findoneBycompyid(String compyid);

	public void updatePic(int id,String pic) ;

}
